lib/active_model/serializer/include_tree.rb

```diff
@@ -0,0 +1,77 @@
+module ActiveModel
+  class Serializer
+    # Description
+
+    class IncludeTree
+      class << self
+        # Builds an IncludeTree from a comma separated list of dot separated paths (JSON API format).
+        # @example `'posts.author, posts.comments.upvotes, posts.comments.author'`
+        #
+        # @param [String] included
+        # @return [IncludeTree]
+        #
```

Comment:
Long `class << self` blocks are hard to read.  Better to just use `def self.method_name`.

I only did it in Adapter because I was `class << Adapter` which is easier to read than `def Adapter.whatever` and a little less surprising, but also I was intending to get rid of it eventually.